src/props/onShippingChange.js

```diff
@@ -100,11 +101,86 @@ export type OnShippingChangeActionsType = {|
     reject : (string) => ZalgoPromise<void>
 |};
 
-export function buildXShippingChangeActions({ orderID, actions, facilitatorAccessToken, buyerAccessToken, partnerAttributionID, forceRestAPI, clientID, experiments } : {| orderID : string, actions : OnShippingChangeActionsType, facilitatorAccessToken : string, buyerAccessToken : ?string, partnerAttributionID : ?string, forceRestAPI : boolean, experiments: Experiments; clientID: string; |}) : XOnShippingChangeActionsType {
+/**
+ * Full matches the following;
+ *  /purchase_units/@reference_id=='default'/amount
+ *  /purchase_units/@reference_id=='default'/shipping/address
+ *  /purchase_units/@reference_id=='default'/shipping/options
+ *  /purchase_units/@reference_id=='d9f80740-38f0-11e8-b467-0ed5f89f718b'/amount
+ */
+const pathPattern = new RegExp(
+    /^\/purchase_units\/@reference_id=='(?:\w|-)*'\/(?:amount|shipping\/(?:options|address))$/
+);
+
+/**
+ *
+ * @param {array} result
+ * @param {{ path: string; }} patch
+ * @returns {array}
+ */
+const sanitizePatch = (rejected, patch) => {
+    const { path } = patch;
+
+    if (!pathPattern.test(path)) {
+        rejected.push(path);
+    }
+    return rejected;
+};
+
+/**
+ *
+ * @param {string} appName
+ * @returns {boolean}
+ */
+export const isWeasley = (appName: string) => appName === 'weasley';
+
+export const logInvalidShippingChangePatches = ({ appName, buyerAccessToken, data, shouldUsePatchShipping }) => {
+    try {
+        if (Array.isArray(data)) {
+            const rejected = data.reduce(sanitizePatch, []);
+            if (rejected.length > 0) {
+                getLogger()
+                .info(`button_shipping_change_patch_data_has_invalid_path_${appName}`, {
+                    appName,
+                    rejected: JSON.stringify(rejected),
+                    hasBuyerAccessToken: String(Boolean(buyerAccessToken)),
+                    shouldUsePatchShipping: String(shouldUsePatchShipping)
+                })
+                .flush();
+            }
+        } else {
+            getLogger()
+            .info('button_shipping_change_patch_data_is_object', {
+                appName,
+                hasBuyerAccessToken: String(Boolean(buyerAccessToken)),
+                shouldUsePatchShipping: String(shouldUsePatchShipping)
+            })
+            .flush();
+        }
+    } catch(err) {
+        getLogger()
+        .error('button_shipping_change_patch_data_logging_failed', {
+            appName,
+            errMessage: JSON.stringify(err),
+            hasBuyerAccessToken: String(Boolean(buyerAccessToken)),
+            shouldUsePatchShipping: String(shouldUsePatchShipping)
+        })
+        .flush();
```

Comment:
Can we remove these `.flush()`s? This should be handled later in the flow.

We get a lot of push back from merchants because of multiple logging requests
```suggestion
                });
            }
        } else {
            getLogger()
            .info('button_shipping_change_patch_data_is_object', {
                appName,
                hasBuyerAccessToken: String(Boolean(buyerAccessToken)),
                shouldUsePatchShipping: String(shouldUsePatchShipping)
            });
        }
    } catch(err) {
        getLogger()
        .error('button_shipping_change_patch_data_logging_failed', {
            appName,
            errMessage: JSON.stringify(err),
            hasBuyerAccessToken: String(Boolean(buyerAccessToken)),
            shouldUsePatchShipping: String(shouldUsePatchShipping)
        });
```

Reply:
Sure; will update